src/main/scala/edu/berkeley/ce/sparkrocks/Block.scala

```diff
@@ -532,41 +584,103 @@ case class Block(center: Array[Double], faces: Seq[Face], generation: Int=0) ext
         allVertices.map { v => v(2) }.sum / allVertices.size
       )
     } else {
-      Array(
-        // Factor of 3 comes from: centroid / (2.0 * (volume/6.0))
-        3.0 * centroidX / totalVolume,
-        3.0 * centroidY / totalVolume,
-        3.0 * centroidZ / totalVolume
-      )
+      val increments: Seq[(Double, Double, Double)] = faces.flatMap { face =>
+        vertices(face).map { faceVertex =>
+          val vertexIndex = vertices(face).indexOf(faceVertex)
+
+          // Tetrahedra are constructed using three vertices at a time, so stop
+          // mapping beyond 3 to last vertex
+          if ( vertexIndex < vertices(face).size - 2 ) {
```

Comment:
Can't do `dropRight` since those elements are used in the calculations as well. I'll use `sliding` like we discussed.